### PR TITLE
Adaptive time stepping in Flow Solver

### PR DESCRIPTION
**Pull request by jbrillon**
PR to incorporate the addition of adaptive time stepping to `flow_solver.cpp` for **unsteady** flow cases. The typical advection based CFL condition has been implemented for the `periodic_turbulent_flow` case. 

NOTE: The default member function definition for `get_adaptive_time_step()` will not be implemented in this PR as its implementation is a bit more involved (must find minimum element size and local polynomial degree for a given `triangulation`); I've opened an `enhancement` issue for this with instructions.

### CODE REVIEW

File: src/flow_solver/flow_solver_cases/flow_solver_case_base.cpp [hunk 1]
```diff
@@ -82,6 +82,14 @@ double FlowSolverCaseBase<dim,nstate>::get_constant_time_step(std::shared_ptr<DG
     return all_param.ode_solver_param.initial_time_step;
 }
 
+template <int dim, int nstate>
+double FlowSolverCaseBase<dim,nstate>::get_adaptive_time_step(std::shared_ptr<DGBase<dim,double>> /*dg*/) const
+{
+    pcout << "Using initial time step in ODE parameters." <<std::endl;
+    pcout << " Base definition for get_adaptive_time_step() has not yet been implemented." <<std::endl;
+    return all_param.ode_solver_param.initial_time_step;
```
Comment (cpethrick) [inline src/flow_solver/flow_solver_cases/flow_solver_case_base.cpp:90]: Maybe a better behaviour would be to abort if the user accidentally sets `adaptive_time_step=true` while this isn't implemented. Shouldn't really matter if the intention is to add this soon, though.

File: src/testing/taylor_green_vortex_restart_check.cpp [hunk 2]
```diff
@@ -80,12 +82,20 @@ int TaylorGreenVortexRestartCheck<dim, nstate>::run_test() const
...
     // Integrate to time at which we stop the run
     std::unique_ptr<FlowSolver::FlowSolver<dim,nstate>> flow_solver_incomplete_run = FlowSolver::FlowSolverFactory<dim,nstate>::select_flow_case(&params_incomplete_run, parameter_handler);
     static_cast<void>(flow_solver_incomplete_run->run());
 
+    const double time_step_at_stop_time = flow_solver_incomplete_run->flow_solver_case->get_constant_time_step(flow_solver_incomplete_run->dg);
+    Parameters::AllParameters params_restart_to_complete_run = reinit_params(false,true,
+                                                                             time_at_which_the_run_is_complete,
+                                                                             time_at_which_we_stop_the_run,
+                                                                             initial_iteration_restart,
+                                                                             0.0,
```
Comment (cpethrick) [inline src/testing/taylor_green_vortex_restart_check.cpp:95]: Use a variable name or add comment - with so many inputs it's hard to remember which this is.

**Review by cpethrick — COMMENTED** [review]
Looks good! I have two minor comments. Should be good to merge in soon, though.

**Review by PranshulThakur — COMMENTED** [review]
Looks great! I have added some comments.

**Review by donovan97 — APPROVED** [review]
Looks good to me.

File: src/flow_solver/flow_solver_cases/flow_solver_case_base.cpp (hunk 1 above)
Comment (jbrillon) [reply src/flow_solver/flow_solver_cases/flow_solver_case_base.cpp]: fixed in most recent commit

File: src/flow_solver/flow_solver.cpp [hunk 3]
```diff
@@ -354,8 +354,19 @@ int FlowSolver<dim,nstate>::run() const
         //----------------------------------------------------
         // Constant time step based on CFL number
```
[the remark this answers is not in the document]
Comment (jbrillon) [reply src/flow_solver/flow_solver.cpp]: fixed in most recent commit

**Review by PranshulThakur — APPROVED** [review]

**Review by cpethrick — APPROVED** [review]
Looks good!

**Pull request merged**